packages/special-pages/messages/release-notes/onUpdate.subscribe.json

```diff
@@ -1,45 +1,195 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
-  "type": "object",
   "title": "UpdateMessage",
   "additionalProperties": false,
   "description": "Message sent from browser when release notes are updated",
-  "properties": {
-    "status": {
-      "description": "Current status of version check",
-      "type": "string",
-      "enum": ["loading", "loaded", "updateReady"]
-    },
-    "currentVersion": {
-      "description": "Current version of the app",
-      "type": "string"
+  "type": "object",
+  "oneOf": [
+    {
+      "type": "object",
+      "required": ["status", "currentVersion", "lastUpdate"],
+      "properties": {
+        "status": {
+          "type": "string",
+          "const": "loading"
+        },
+        "currentVersion": {
+          "description": "Current version of the app",
+          "type": "string"
+        },
+        "lastUpdate": {
+          "description": "Timestamp of last check for version updates",
+          "type": "number"
+        }
+      }
     },
-    "latestVersion": {
-      "description": "Latest version of the app. May be the same as currentVersion",
-      "type": "string"
+    {
+      "type": "object",
+      "required": ["status", "currentVersion", "latestVersion", "lastUpdate"],
+      "properties": {
+        "status": {
+          "type": "string",
+          "const": "loaded"
+        },
+        "currentVersion": {
+          "description": "Current version of the app",
+          "type": "string"
+        },
+        "latestVersion": {
+          "description": "Latest version of the app. May be the same as currentVersion",
+          "type": "string"
+        },
+        "lastUpdate": {
+          "description": "Timestamp of last check for version updates",
+          "type": "number"
+        },
+        "releaseTitle": {
+          "description": "Name of the current release (e.g. April 26 2024)",
+          "type": "string"
+        },
+        "releaseNotes": {
+          "description": "Array containing notes for the latest release",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "releaseNotesPrivacyPro": {
+          "description": "Array containing Privacy Pro notes for the latest release",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
     },
-    "lastUpdate": {
-      "description": "Timestamp of last check for version updates",
-      "type": "number"
+    {
+      "type": "object",
```

Comment:
if you give these objects a `title` field - the generated types will be improved (look at the TS this generates)

Reply:
Yeah, I had that ready to go but hadn’t pushed yet as I halfway through a refactor. Done now. Thanks!